src/include/extension/extension.h

```diff
@@ -53,14 +53,14 @@ struct ExtensionSourceUtils {
 };
 
 template<typename T>
-void addFunc(transaction::Transaction* transaction, main::Database& database, std::string name,
-    catalog::CatalogEntryType functionType, bool isInternal = false) {
+void addFunc(main::Database& database, std::string name, catalog::CatalogEntryType functionType,
+    bool isInternal = false) {
     auto catalog = database.getCatalog();
-    if (catalog->containsFunction(transaction, name, isInternal)) {
+    if (catalog->containsFunction(&transaction::DUMMY_TRANSACTION, name, isInternal)) {
```

Comment:
Can `containsFunction` and `addFunction` not work without accepting a transaction?

Reply:
From the interface design perspective, they should always require a transaction.